listings/applications/components/src/switchable.cairo

```diff
@@ -49,3 +49,85 @@ pub mod switchable_component {
 // ANCHOR_END: component
 
 
+#[starknet::contract]
+mod MockContract {
+    use super::switchable_component;
+
+    component!(path: switchable_component, storage: switchable, event: SwitchableEvent);
+
+    #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        switchable: switchable_component::Storage,
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        #[flat]
+        SwitchableEvent: switchable_component::Event
+    }
+
+    #[abi(embed_v0)]
+    impl SwitchableImpl = switchable_component::Switchable<ContractState>;
+}
+
+#[cfg(test)]
+mod test {
+    use super::MockContract;
+    use super::switchable_component::{Event, SwitchEvent};
+    use super::{ISwitchableDispatcher, ISwitchableDispatcherTrait};
+    use starknet::{syscalls::deploy_syscall, contract_address_const, ContractAddress};
+    use starknet::SyscallResultTrait;
+
+    fn deploy_switchable() -> (ISwitchableDispatcher, ContractAddress) {
+        let (address, _) = deploy_syscall(
+            MockContract::TEST_CLASS_HASH.try_into().unwrap(), 0, array![].span(), false
+        )
+            .unwrap_syscall();
+        (ISwitchableDispatcher { contract_address: address }, address)
+    }
+
+    #[test]
+    fn test_constructor() {
+        let (switchable, _) = deploy_switchable();
+        assert_eq!(switchable.is_on(), false);
+    }
+
+    #[test]
+    fn test_switch() {
+        let (switchable, _) = deploy_switchable();
+        switchable.switch();
+        assert_eq!(switchable.is_on(), true);
+    }
+
+    #[test]
+    fn test_multiple_switches() {
+        let (switchable, _) = deploy_switchable();
+        switchable.switch();
+        switchable.switch();
+        switchable.switch();
+        assert_eq!(switchable.is_on(), true);
+    }
+
+    #[test]
+    fn test_toggle_back_and_forth() {
+        let (switchable, _)= deploy_switchable();
+        switchable.switch();
+        assert_eq!(switchable.is_on(), true);
+        switchable.switch();
+        assert_eq!(switchable.is_on(), false);
+        switchable.switch();
+        assert_eq!(switchable.is_on(), true);
+    }
+
+    #[test]
+    fn test_switch_event() {
+        let (switchable, contract_address) = deploy_switchable();
+        switchable.switch();
+        assert_eq!(
+            starknet::testing::pop_log(contract_address),
+            Option::Some(Event::SwitchEvent(SwitchEvent { }))
```

Comment:
```suggestion
            Option::Some(MockContract::Event::SwitchableEvent(SwitchEvent { }.into()))
```

Reply:
This test can also be merged with `test_switch`, since they are testing the same flow.